src/components/molecules/ChatPoll/ChatPoll.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useCallback, useMemo } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPoll } from "@fortawesome/free-solid-svg-icons";
+
+import { PollMessage, BaseMessageToDisplay, Vote } from "types/chat";
+
+import { WithId } from "utils/id";
+
+import { ChatMessageInfo } from "components/atoms/ChatMessageInfo";
+import { TextButton } from "components/atoms/TextButton";
+import Button from "components/atoms/Button";
+
+import { useUser } from "hooks/useUser";
+
+import "./ChatPoll.scss";
+
+export interface ChatPollProps {
+  pollData: WithId<BaseMessageToDisplay<PollMessage>>;
+  deletePoll: (pollId: string) => void;
+  voteInPoll: (votes: Vote[], pollId: string) => void;
+}
+
+export const ChatPoll: React.FC<ChatPollProps> = ({
+  pollData,
+  voteInPoll,
+  deletePoll,
+}) => {
+  const { user } = useUser();
+
+  const userId = user?.uid;
+
+  const { id, poll, votes, isMine } = pollData;
+  const { questions, topic, canBeDeleted } = poll;
+
+  const isVoted = userId
+    ? votes.map(({ userId }) => userId).includes(userId)
+    : false;
+
+  const isDeleted = isMine && canBeDeleted;
+
+  const containerStyles = classNames("ChatPoll", {
+    "ChatPoll--me": isMine,
+  });
+
+  const formattedQuestions = questions
+    .map((item) => ({
+      ...item,
+      count: votes.length
+        ? Math.floor(
+            (votes.filter(({ questionId }) => questionId === item.id).length /
+              votes.length) *
+              100
+          )
+        : 0,
+    }))
+    .sort((a, b) => b.count - a.count);
```

Comment:
The naming doesn't explain what it does. + the logic should be moved to a function

Reply:
> the logic should be moved to a function

Or potentially just simplified. What's the need for using the ternary to check `votes.length`?

If `votes.length` is 0, then the `.filter` will basically become a noop, which means that `votes.filter().length` will just return `0`. Which will end up being `0 / 0 * 100`, which will give a divide by 0 error.

So yeah.. maybe not simplified. Function sounds good.

Reply:
fixed in bbbbdb38939db532c7e7dd219a37840272a8bf37

Well, to render poll results I definitely need a prop (renamed it from `count` -> `share`) that stores the `share` (percentage from all polling results) of the particular question. 

I'm using a separate `calculateShare` method to make all calculations inside it.

it's needed to sort all questions DESC and correctly display their share via background-color